pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -185,6 +180,34 @@ func (r *reconciler) createBuildsForArchitectures(ctx context.Context, mabc *v1.
 	return nil
 }
 
+func (r *reconciler) handlePushImageWithManifest(mabc *v1.MultiArchBuildConfig, targetImageRef string) func(mabcToMutate *v1.MultiArchBuildConfig) {
+	if cond := getCondition(mabc, PushImageManifestDone); cond != nil {
```

Comment:
This shouldn't be happening here.

Reply:
Where is should happen?

Reply:
The reconciler should determine which of the operation should run based on the conditions.

Reply:
By reconciler you mean the `handleMultiArchBuildConfig` function? Because `handlePushImageWithManifest` is a reconciler's method too.

Reply:
All I am saying is that you don't need to run the `handlePushImageWithManifest` method if there is no need to do it.

Reply:
There is no other operation that updates the conditions. You can just open a switch statement for the latest condition and act accordingly. The `getCondition` now is useless since you don't need the condition for any reason rather than checking if the `PushManifestDone` is done.

Reply:
Why a switch statement then? I could just check the last one in that case.